Sources/HipMobileUI/ViewModels/Pages/UserRatingPageViewModel.cs

```diff
@@ -0,0 +1,369 @@
+        private string star3BarCount;
+        private string star4BarCount;
+        private string star5BarCount;
+
+        private string ratingAverage;
+        private string ratingStars;
+        private string ratingCount;
+
+        private int rating = 0;
+        #endregion
+
+
+        public UserRatingPageViewModel(Exhibit exhibit)
+        {
+            Exhibit = exhibit;
+            Headline = exhibit.Name;
+
+            var imageData = exhibit.Image.GetDataBlocking();
+            Image = imageData != null ? ImageSource.FromStream(() => new MemoryStream(imageData)) : ImageSource.FromStream(() => new MemoryStream(BackupData.BackupImageData));
+
+            SetUserRatingUi();
+            SendRatingCommand = new Command(SendUserRating);
+            SelectStarCommand = new Command(OnSelectStar);
+
+        }
+        private async void SetUserRatingUi()
+        {
+            UserRating userRating = await IoCManager.Resolve<IUserRatingManager>().GetUserRating(exhibit);
+            SetRatingAverageAndCount(userRating.Average, userRating.Count);
+            SetRatingBars(userRating.RatingTable, userRating.Count);
+            SetRatingStars();
+        }
+
+        private void SetRatingAverageAndCount(double average, int count)
+        {
+            string stars = "";
+            for (int i = 1; i <= 5; i++)
+            {
+                if (average >= i)
+                    stars += "★";
+                else
+                    stars += "☆";
+            }
+            if (count > 0)
+                RatingAverage = average.ToString("0.#");
+            else
+                RatingAverage = "-";
+            RatingStars = stars;
+            RatingCount = count.ToString() + " " + Strings.UserRating_Rate_Count;
+        }
+
+
+        private void SetRatingBars(Dictionary<int, int> ratingTable, int count)
+        {
+            Star5BarCount = ratingTable[5].ToString();
+            Star4BarCount = ratingTable[4].ToString();
+            Star3BarCount = ratingTable[3].ToString();
+            Star2BarCount = ratingTable[2].ToString();
+            Star1BarCount = ratingTable[1].ToString();
+            Star5Bar = CalculateBarProportion(ratingTable[5], count);
+            Star4Bar = CalculateBarProportion(ratingTable[4], count);
+            Star3Bar = CalculateBarProportion(ratingTable[3], count);
+            Star2Bar = CalculateBarProportion(ratingTable[2], count);
+            Star1Bar = CalculateBarProportion(ratingTable[1], count);
+        }
+
+        private GridLength CalculateBarProportion(double value, double totalCount)
+        {
+            if (value == 0)
+                return new GridLength(0, GridUnitType.Absolute);
+            if (value == totalCount)
+                return new GridLength(short.MaxValue, GridUnitType.Star);
+            double prop = (1 / (1 - value / totalCount) - 1);
+            return new GridLength(prop, GridUnitType.Star);
+        }
+
```

Comment:
Here and after other methods are extra blank lines. There should only be one blank line in between class members.